app/src/main/java/org/p2p/wallet/newsend/SendFeeRelayerManager.kt

```diff
@@ -145,10 +147,10 @@ class SendFeeRelayerManager(
                         feeRelayerFee = feeState.feeInSol
                     )
                     currentState = UpdateFee(solanaFee = solanaFee, feeLimitInfo = feeLimitInfo)
-                    sendInteractor.setFeePayerToken(solToken)
+                    sendInteractor.switchFeePayerToSol(solToken)
                 }
                 is FeeCalculationState.Success -> {
-                    sendInteractor.setFeePayerToken(sourceToken)
+                    sendInteractor.setFeePayerToken(feePayer)
```

Comment:
i've tried to pass feePayer token instead of sourceToken because it doesn't make sense to always pass source token